packages/rum-core/src/domain/rumEventsCollection/view/interactionCountPolyfill.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * performance.interactionCount polyfill
+ *
+ * The interactionCount is an integer which counts the total number of distinct user interactions,
+ * for which there was a unique interactionId.
+ *
+ * The interactionCount polyfill is an estimate based on a convention specific to Chrome
+ *
+ * Reference implementation: https://github.com/GoogleChrome/web-vitals/blob/main/src/lib/polyfills/interactionCountPolyfill.ts
+ */
+
+import { monitor } from '@datadog/browser-core'
+import type { BrowserWindow, RumEventTiming, RumPerformanceObserver } from '../../../browser/performanceCollection'
+
+let observer: RumPerformanceObserver | undefined
+
+let interactionCountEstimate = 0
+let minKnownInteractionId = Infinity
+let maxKnownInteractionId = 0
+
+export function initInteractionCountPolyfill() {
+  if ('interactionCount' in performance || observer) {
+    return
+  }
+
+  observer = new (window as BrowserWindow).PerformanceObserver(
+    monitor((entries: PerformanceObserverEntryList) => {
+      entries.getEntries().forEach((e) => {
+        const entry = e as unknown as RumEventTiming
+
+        if (entry.interactionId) {
+          minKnownInteractionId = Math.min(minKnownInteractionId, entry.interactionId)
+          maxKnownInteractionId = Math.max(maxKnownInteractionId, entry.interactionId)
+
+          interactionCountEstimate = maxKnownInteractionId ? (maxKnownInteractionId - minKnownInteractionId) / 7 + 1 : 0
```

Comment:
About the `/ 7 + 1`, from the original Google [PR](https://github.com/GoogleChrome/web-vitals/pull/213) description, we don't have much information besides `interactionCount is an estimate based on a convention specific to Chrome`

Good suggestion :)

Reply:
`specific to Chrome` that's not great 😅 For now, this "polyfill" (and the whole INP collection) will only work on Chrome, but if/when other browsers support `interactionId`, this could be an issue. Hopefully they'll implement `performance.interactionCount` at the same time, so we won't need that polyfill.

Could you add a code comment to that PR to justify the approach here?

Reply:
I added more detail in the comment section https://github.com/DataDog/browser-sdk/blob/85e1a329850922ef644b1ace9ff959edc546db09/packages/rum-core/src/domain/rumEventsCollection/view/interactionCountPolyfill.ts#L7-L9